.github/workflows/PyLintFixOnPush.yaml

```diff
@@ -0,0 +1,39 @@
+name: Linter
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  cpu:
+    name: "PyLint test and Fix"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        python-version: ['3.10']
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint pyink
+    - name: Analysing the code with pylint in Maxtext
+      run: |
+         pylint MaxText/ --recursive=y \
```

Comment:
Can we move these pylint runs under the cpu: section of `.github/workflows/UnitTests.yml`, that way we can avoid having a seperate yml file for this.

Reply:
I create this separate file because the UnitTest runs on 
```
on:
  pull_request:
  push:
    branches: [ "main" ]
```
and I want the pylint check to run on every push on every brnach. I think we don't want UnitTest.yaml to run on all pushes and all branches. Happy to change if that's the case

Reply:
Then I think we can remove the cpu tests from that file and also have pytype run on all branches in this workflow.

Reply:
After offline discussion moved all cpu tests to run on each push to remove duplicate tests.